### ISSUE TITLE
python3.8 complains about circular import

### ISSUE THREAD

**Issue by francesco-ballarin** [FEniCS/dolfinx#2561]
I am working on restoring my FEM on Colab workflow, which has been broken due to https://github.com/fem-on-colab/fem-on-colab/issues/19.
While testing a new workaround, I now face the following

```
$ python3 -c 'import dolfinx'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/dolfinx/__init__.py", line 29, in <module>
    from dolfinx import common
  File "/usr/local/lib/python3.8/site-packages/dolfinx/common.py", line 11, in <module>
    from dolfinx import cpp as _cpp
ImportError: cannot import name 'cpp' from partially initialized module 'dolfinx' (most likely due to a circular import) (/usr/local/lib/python3.8/site-packages/dolfinx/__init__.py)
```

My workflow has been broken for a couple of weeks, so any of https://github.com/FEniCS/dolfinx/pull/2532 https://github.com/FEniCS/dolfinx/pull/2539 https://github.com/FEniCS/dolfinx/pull/2540 https://github.com/FEniCS/dolfinx/pull/2457 https://github.com/FEniCS/dolfinx/pull/2555 https://github.com/FEniCS/dolfinx/pull/2547 https://github.com/FEniCS/dolfinx/pull/2560 (i.e., PR that have touched `python/dolfinx`) could have been the cause of the issue. 
I would probably blame https://github.com/FEniCS/dolfinx/pull/2555: any idea why this issue would only affect python 3.8, but not later versions present on CI?

**Comment by garth-wells** [comment]
I have seen this as as symptom when there is something wrong with the compiled binary.

**Comment by garth-wells** [comment]
This might fix the issue: #2567.

It fixes the same error message with Spack. Problem was the compiled `cpp` module was not being installed in the (non-standard) location of the other dolfinx Python modules.

**Comment by francesco-ballarin** [comment]
Thanks, I'll test this tomorrow and close the issue if all goes well.
